cpp/src/unary/math_ops.cu

```diff
@@ -559,6 +574,19 @@ struct FixedPointOpDispatcher {
   }
 };
 
+inline bool is_signed_numeric(type_id id)
+{
+  switch (id) {
+    case type_id::INT8:
+    case type_id::INT16:
+    case type_id::INT32:
+    case type_id::INT64:
+    case type_id::FLOAT32:
+    case type_id::FLOAT64: return true;
+    default: return false;
+  }
+}
+
```

Comment:
We have a bunch of these in `utilities/type_traits.hpp`. I think in this case you want `cudf::is_numeric(input.type()) && !cudf::is_unsigned(input.type())`

Reply:
Oh cool, thanks

Reply:
@davidwendt just added some traits around this. I think there may be one called `is_signed` which would be better here. Please check me.